derive_miniconf/src/field.rs

```diff
@@ -0,0 +1,93 @@
+use super::{
+    attributes::{AttributeParser, MiniconfAttribute},
+    TypeDefinition,
+};
+use syn::parse_quote;
+
+pub struct StructField {
+    pub field: syn::Field,
+    pub deferred: bool,
+}
+
+impl StructField {
+    pub fn new(field: syn::Field) -> Self {
+        let deferred = field
+            .attrs
+            .iter()
+            .filter(|attr| attr.path.is_ident("miniconf"))
+            .map(|attr| AttributeParser::new(attr.tokens.clone()).parse())
+            .any(|x| x == MiniconfAttribute::Defer);
```

Comment:
It seems like any `Array` type should also be deferred by-default. It's relatively rare that a user would want to set the entire array in a single invocation.